### PR TITLE
user-presence resolver seems ok

### PR DESCRIPTION
**Pull request by Chicanoss**
First version of the resolver and spec of Userpresence.

I made a test for each methods.
I tried to follow the norm.

Waiting for reviews !

### CODE REVIEW

File: backend/src/user-presence/user-presence.module.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 import { PrismaModule } from 'src/prisma/prisma.module'
 import { UserPresenceService } from './user-presence.service'
+import { UserPresence } from './entities/user-presence.entity'
+import { UserPresenceResolver } from './user-presence.resolver'
 
 @Module({
   imports: [PrismaModule],
-  exports: [UserPresenceService],
-  providers: [UserPresenceService]
+  exports: [UserPresenceService, UserPresence],
```
Comment (kbarbry) [inline backend/src/user-presence/user-presence.module.ts:9]: pas necessaire d'export UserPresence, a supprimer

File: backend/src/user-presence/user-presence.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  ): Promise<UserPresence> {
+    return this.userpresenceService.disconnected(id)
+  }
+
+  //**************************************************//
+  //  QUERY
+  //**************************************************//
+
+  @Query(() => Boolean)
+  async isConnected(@Args('id', { type: () => String }) id: string) {
```
Comment (kbarbry) [inline backend/src/user-presence/user-presence.resolver.ts:35]: pas de valeur de retour 
```ts
: Promise<xxx>
```

File: backend/src/user-presence/user-presence.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return this.userpresenceService.isConnected(id)
+  }
+
+  @Query(() => UserPresence)
+  async findOne(@Args('id', { type: () => String }) id: string) {
```
Comment (kbarbry) [inline backend/src/user-presence/user-presence.resolver.ts:40]: pas de valeur de retour 
```ts
: Promise<xxx>
```

File: backend/src/user-presence/user-presence.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    return this.userpresenceService.findOne(id)
+  }
+
+  @Query(() => UserPresence)
+  async findLastByUserId(@Args('id', { type: () => String }) id: string) {
```
Comment (kbarbry) [inline backend/src/user-presence/user-presence.resolver.ts:45]: pas de valeur de retour 
```ts
: Promise<xxx>
```

File: backend/src/user-presence/user-presence.resolver.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    return this.userpresenceService.findLastByUserId(id)
+  }
+
+  @Query(() => UserPresence)
+  async findAllByUserId(@Args('id', { type: () => String }) id: string) {
```
Comment (kbarbry) [inline backend/src/user-presence/user-presence.resolver.ts:50]: pas de valeur de retour 
```ts
: Promise<xxx>
```

**Review by kbarbry — CHANGES_REQUESTED** [review]
very clean overall just a few non major mistakes

**Review by kbarbry — APPROVED** [review]

**Review by Isioblast — CHANGES_REQUESTED** [review]

**Pull request merged**